### DISCUSSION TITLE
Motor Thrust Setting

### DISCUSSION THREAD

**Discussion by yueminm** [bitcraze/discussions#114] [Q&A] [answered]
Hi!

I'm wondering how to convert the value of `ithrust` to the actual force the motor generates. 

To my understanding, `ithrust = 65536` maps to `motor thrust = 60 grams`, however, when I set `ithrust` to be at 8000 for all four motors, which based on this conversion would be 29 grams of total thrust, slightly over the crazyflie mass, it could not take off. I've tested with a joystick, when the crazyflie took off `ithrust` for each motor was around 30000, much higher than I expected. 

I've also looked at this website: [https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/pwm-to-thrust/](url), it says `RPM = 23882` maps to `thrust = 57.9g` and `PWM = 93.75%`, but the maximum of `ithrust` is 65536, and the actual test also showed that this value could reach more than 60000, so I'm assuming this command parameter `ithrust` is different from RPM.

Can you clarify what this parameter `ithrust` is and how to convert it to the actual thrust in gram? Actually I'm converting thrust force required to `ithrust` in my controller so it would be better to know how to convert actual thrust to `ithrust`.

Thanks!

**Comment by tobbeanton** [comment, marked as answer]
I think the misunderstanding is that 60 grams is for all motors summed. Thus 29 grams of trust would equal ~31700 pwm.

The function which is also described in the [code](https://github.com/bitcraze/crazyflie-firmware/blob/bdf038679747fe862531d49548ad0f9953280065/src/drivers/src/motors.c#L128) is just an approximation. It also takes the voltage level into account which has a big effect on the output thrust.

If you want to use thrust in grams as input you could rewrite the function and instead input the thrust in grams directly:
```
static uint16_t motorsForceToPWM(float thrustGrams)
{
  float supply_voltage = pmGetBatteryVoltage();
  /*
   * A LiPo battery is supposed to be 4.2V charged, 3.7V mid-charge and 3V
   * discharged.
   */
// float thrust = ((float) ithrust / 65536.0f) * 60; // Remove the PWM to gram conversion
  float volts = -0.0006239f * thrust * thrust + 0.088f * thrustGrams;
  float percentage = volts / supply_voltage;
  percentage = percentage > 1.0f ? 1.0f : percentage;
  return percentage * UINT16_MAX;
}
```

I'm not that sure how accurate it will be though. You should probably test this.

**Reply by yueminm** [reply]
Thanks!
